File: AN_2/SEM_1/MAP/lab9/lab9/Model/MessageTask.cs
using System;

namespace Model
{
    public class MessageTask : Tasks
    {
        private string message;
        private string from;
        private string to;
        private DateTime date;
        private static readonly string dateFormat = "yyyy-MM-dd HH:mm";

        public MessageTask(string taskId, string description, string message, string from, string to, DateTime date)
            : base(taskId, description)
        {
            this.message = message;
            this.from = from;
            this.to = to;
            this.date = date;
        }

        public override void Execute()
        {
            Console.WriteLine($"{message} from: {from} to: {to} {date.ToString(dateFormat)}: ");
        }

        public override string ToString()
        {
            return $"id = {Id} | description = {Description} | message = {message} | from = {from} | to = {to} | date = {date.ToString(dateFormat)}";
        }
    }
}

File: AN_2/SEM_1/MAP/lab2/src/model/MessageTask.java
package model;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class MessageTask extends Task
{
    private String message;
    private String from;
    private String to;
    private LocalDateTime date;
    private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");

    public MessageTask(String taskId, String description, String message, String from, String to, LocalDateTime date)
    {
        super(taskId, description);
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }

    @Override
    public void execute()
    {
        System.out.println(message + " from: " + from + " to: " + to + date.format(dateFormatter) + ": ");
    }

    @Override
    public String toString()
    {
        return "id = " + getId() +
                " | description = " + getDescription() +
                " | message = " + message +
                " | from = " + from +
                " | to = " + to +
                " | date = " + date.format(dateFormatter);
    }
}
